.github/actions/maven-build-and-tag/action.yml

```
@@ -240,13 +257,19 @@ runs:
         echo "testcontainers.reuse.enable=true" > ~/.testcontainers.properties
         echo "TESTCONTAINERS_RYUK_DISABLED=true" >> $GITHUB_ENV
 
+    - name: Set SONAR_SCANNER_OPTS
+      if: ${{ inputs.sonar-token != '' && github.actor != 'dependabot[bot]' }}
```

Im not sure about setting this here on this level. How about adding simple true false switch from inputs and check if we want to run this step in upstream workflow. @gionn

---

```suggestion
            if: ${{ inputs.enable-sonar-scan && github.actor != 'dependabot[bot]' }}
```
like this?

---

I need the check on the dependabot because I don't want to execute a sonar scan in the dependabot pr

---

no need to explicitly checks for github.actor, dependabot would not have access to sonar-token anyway.

I would just have:

```suggestion
      if: ${{ inputs.sonar-token != '' }}
```

boolean help in being explicit but in this case we have a secret and really make it straight-forward to use just that